backend/user-service/controller/user-controller.ts

```diff
@@ -334,6 +341,87 @@ export async function updateUser(
   }
 }
 
+export const sendResetPasswordMail = async (
+  req: Request,
+  res: Response
+): Promise<Response> => {
+  try {
+    const { email } = req.body;
+    const user = await _findUserByEmail(email);
+
+    if (!user) {
+      return res.status(404).json({ message: `User not found` });
+    }
+
+    const emailToken = crypto.randomBytes(16).toString("hex");
+    await redisClient.set(email, emailToken, { EX: 60 * 5 }); // expire in 5 minutes
```

Comment:
do we need to differentiate the token sent for email verification and password reset?